tests/TestCase/ORM/Association/HasManyTest.php

```diff
@@ -1121,6 +1121,42 @@ public function testSaveReplaceSaveStrategyDependent()
         $this->assertFalse($authors->Articles->exists(['id' => $articleId]));
     }
 
+    /**
+     * Test that the associated entities are unlinked and deleted when they are dependent
+     * when associated entities array is indexed by string keys
+     *
+     * @return void
+     */
+    public function testSaveReplaceSaveStrategyDependentWithStringKeys()
+    {
+        $authors = $this->getTableLocator()->get('Authors');
+        $authors->hasMany('Articles', ['saveStrategy' => HasMany::SAVE_REPLACE, 'dependent' => true]);
+
+        $entity = $authors->newEntity([
+            'name' => 'mylux',
+            'articles' => [
+                ['title' => 'One Random Post', 'body' => 'The cake is not a lie'],
+                ['title' => 'Another Random Post', 'body' => 'The cake is nice'],
+                ['title' => 'One more random post', 'body' => 'The cake is forever']
+            ]
+        ], ['associated' => ['Articles']]);
+
+        $entity = $authors->save($entity, ['associated' => ['Articles']]);
```

Comment:
saveOrFail() ?

Reply:
I've modified test case based on your review. It was just copy-paste of test case above this one with few modifications, so I didn't look much at those lines.